pyramid_swagger/api.py

```diff
@@ -101,26 +101,268 @@ def view_for_api_declaration(request):
+    modify the server specifications.
+
+    :param spec: Swagger Spec
+    :type spec: bravado_core.spec.Spec
+    :param target: URL of the target that has to be formatted
+    :type target: str
+    :param current_path: local or remote base path used for the absolute URL
+        generation. It represent the path of the resource in which is defined
+        the target/reference.
+    :type current_path: str
+    :return: url associated to the target resource
+    """
+    if target == '':
+        raise ValueError('Empty target')
+
+    target_url = _get_absolute_link(spec, target, current_path)
+    target_scheme = 'file' if target_url.scheme == '' else target_url.scheme
+
+    if target_scheme == 'file':
+        spec_dir = os.path.dirname(urlparse(spec.origin_url).path)
+        # Hiding of the internal server paths information.
+        # A path relative to the swagger.{json,yaml} is returned
+        target_url = urlparse('{path}#{fragment}'.format(
+            path=_relpath(target_url.path, spec_dir),
+            fragment=target_url.fragment,
+        ))
+
+    if target_scheme in ['file', 'http', 'https']:
+        return target_url
+    else:  # Handle the case of an unknown target scheme
+        raise ValueError(
+            'Invalid target: {target}'.format(target=target)
+        )
+
+
+def _marshal_target(target_url):
+    target_scheme = target_url.scheme
+    if len(target_url.path) > 0 and \
+            target_scheme in ['', 'file', 'http', 'https']:
+        return _low_level_translate(
+            '{scheme}://{host}{path}#{fragment}'.format(
+                scheme=target_url.scheme if len(target_scheme) > 0 else 'file',
+                host=target_url.hostname if target_url.hostname else '',
+                path=target_url.path,
+                fragment=target_url.fragment,
+            ))
+    else:
+        raise ValueError(
+            'Invalid target: {target}'.format(target=urlunparse(target_url))
+        )
+
+
+def _unmarshal_target(marshaled_target):
+    if any(marshaled_target.startswith(x) for x in
+           ('file.', 'http.', 'https.')):
+        result = _low_level_translate(marshaled_target, is_marshaling=False)
+        # Remove the file:// scheme to allow the use of relative paths
+        if not result.startswith('file:///'):
+            result = result.replace('file://', '')
+        return result
+    else:
+        raise ValueError(
+            'Invalid marshaled object: {target}'.format(
+                target=marshaled_target
+            )
+        )
+
+
+def is_a_definition(json_path):
+    # definition is possible if:
+    # json_path = ['definitions']
+    # json_path = [ ..., 'schema']
+    # json_path = [ ..., 'items']
+    # json_path = [ ..., 'allOf']
+    return json_path == ['definitions'] or \
+           json_path[-1] in ['schema', 'items', 'allOf']
+
+
+def resolve_ref(spec, url, json_path, file_path, defs_dict):
```

Comment:
Have you done some basic speed tests before and after? The additional logic hasn't made things too much slower I hope?